Index: boost-1.49.0/src/boost/gil/extension/io/png_io_private.hpp
===================================================================
--- boost-1.49.0.orig/src/boost/gil/extension/io/png_io_private.hpp	2010-11-29 16:08:00.000000000 +0100
+++ boost-1.49.0/src/boost/gil/extension/io/png_io_private.hpp	2012-08-02 11:17:29.587631803 +0200
@@ -225,7 +225,7 @@
         int bit_depth, color_type, interlace_type;
         png_get_IHDR(_png_ptr, _info_ptr,
                      &width, &height,&bit_depth,&color_type,&interlace_type,
-                     int_p_NULL, int_p_NULL);
+                     NULL, NULL);
         io_error_if(((png_uint_32)view.width()!=width || (png_uint_32)view.height()!= height),
                     "png_reader_color_convert::apply(): input view size does not match PNG file size");
         switch (color_type) {
